tests/mock/test_mock_session_gone.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2017-2019 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#include <string.h>
+#include "../utils/test_setup.h"
+#include "../utils/mock_setup.h"
+
+void test_session_gone_during_query(void **unused) {
+    setup_mock_login_standard();
+
+    SF_CONNECT *sf = setup_snowflake_connection();
+    SF_STATUS status = snowflake_connect(sf);
+    if (status != SF_STATUS_SUCCESS) {
+        dump_error(&(sf->error));
+    }
+    assert_int_equal(status, SF_STATUS_SUCCESS);
+
+    /* query */
+    setup_mock_query_session_gone();
+    SF_STMT *sfstmt = snowflake_stmt(sf);
+    status = snowflake_query(sfstmt, "select 1;", 0);
+    if (status != SF_STATUS_ERROR_CONNECTION_NOT_EXIST) {
+        dump_error(&(sfstmt->error));
+    }
+    assert_int_equal(status, SF_STATUS_ERROR_CONNECTION_NOT_EXIST);
```

Comment:
This is where we check to make sure that an error is returned to the user if we get a session gone error code.